### PR TITLE
Created CustomerServiceImpl and tests for him

### PR DESCRIPTION
**Pull request by Nikita-Exeption**

### CODE REVIEW

File: src/main/java/org/nikita/chiken_bell/core/service/CustomerService.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package org.nikita.chiken_bell.core.service;
+
+import org.nikita.chiken_bell.core.entity.Customer;
+
+import java.util.Collection;
+import java.util.Optional;
+
+public interface CustomerService {
+
+    Customer create(String name, String phone);
```
Comment (zxRasp) [inline src/main/java/org/nikita/chiken_bell/core/service/CustomerService.java:10]: We need only one "create" method.

File: src/main/java/org/nikita/chiken_bell/core/service/CustomerService.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    Customer create(String name, String phone, String address);
+
+    Customer update(String id, String name, String phone);
```
Comment (zxRasp) [inline src/main/java/org/nikita/chiken_bell/core/service/CustomerService.java:14]: We need only one "update" method

File: src/main/java/org/nikita/chiken_bell/core/service/CustomerService.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    Customer update(String id, String name, String phone, String address);
+
+    Optional<Customer> getById(String id);
+
```
Comment (zxRasp) [inline src/main/java/org/nikita/chiken_bell/core/service/CustomerService.java:19]: We also need "findByPhone" method.

File: src/main/java/org/nikita/chiken_bell/core/service/impl/CustomerServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    @Override
+    public Optional<Customer> getById(String id) {
+        return customers.stream().filter(x -> x.getId().equals(id)).findFirst();
+    }
+
+    @Override
+    public Collection<Customer> getAll() {
+        return customers;
```
Comment (zxRasp) [inline src/main/java/org/nikita/chiken_bell/core/service/impl/CustomerServiceImpl.java:52]: Should return unmodifiable copy, not original collection!

File: src/main/java/org/nikita/chiken_bell/core/service/impl/CustomerServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+import java.util.Set;
+
+public class CustomerServiceImpl implements CustomerService {
+
+    private final Set<Customer> customers = new HashSet<>();
+
+    @Override
+    public Customer create(String name, String phone) {
+        Customer customer = new Customer(name, phone);
+        customers.add(customer);
```
Comment (zxRasp) [inline src/main/java/org/nikita/chiken_bell/core/service/impl/CustomerServiceImpl.java:19]: At this point service must check, that customer's phone is unique.

File: src/main/java/org/nikita/chiken_bell/core/service/impl/CustomerServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        Customer customer = new Customer(name, phone, address);
+        customers.add(customer);
+        return customer;
+    }
+
+    @Override
+    public Customer update(String id, String name, String phone) {
+        Customer customer = getById(id).orElseThrow(CustomerNotFoundException::new);
+        customer.setName(name);
+        customer.setPhone(phone);
```
Comment (zxRasp) [inline src/main/java/org/nikita/chiken_bell/core/service/impl/CustomerServiceImpl.java:34]: At this point service must check, that customer's phone is unique.

**Review by zxRasp — CHANGES_REQUESTED** [review]
Please fix issues.

File: src/main/java/org/nikita/chiken_bell/core/service/CustomerService.java (hunk 2 above)
Comment (Nikita-Exeption) [reply src/main/java/org/nikita/chiken_bell/core/service/CustomerService.java:14]: Which of them

File: src/main/java/org/nikita/chiken_bell/core/service/CustomerService.java (hunk 1 above)
Comment (Nikita-Exeption) [reply src/main/java/org/nikita/chiken_bell/core/service/CustomerService.java:10]: Which of them

File: src/main/java/org/nikita/chiken_bell/core/service/impl/CustomerServiceImpl.java (hunk 5 above)
Comment (Nikita-Exeption) [reply src/main/java/org/nikita/chiken_bell/core/service/impl/CustomerServiceImpl.java:19]: ok

**Pull request closed without merging**